specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json

```diff
@@ -796,6 +796,423 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/BotServices/listServiceProviders": {
+      "post": {
+        "tags": [
+          "ListServiceProviders"
+        ],
+        "description": "Lists the available Service Providers for creating Connection Settings",
+        "operationId": "BotConnections_ListServiceProviders",

```

Comment:
It looks like there are two Operation Group names present here: "BotConnections" and "BotConnection" this will result in two separate client types being generated. Is this the intended behavior?

Reply:
I found another one "BotToken", should that be dangled off of a separate client?

Reply:
removed all "connections" and token references and updated them to be connection to keep things consistent